Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/FareCalculator.hs

```diff
@@ -372,3 +377,27 @@ isTimeWithinBounds startTime endTime time =
       let midnightBeforeTimeleap = TimeOfDay 23 59 60
       (startTime < time && time < midnightBeforeTimeleap) || (midnight <= time && time < endTime)
     else startTime < time && time < endTime
+
+isNightAllowanceApplicable :: Maybe DFP.NightShiftBounds -> UTCTime -> UTCTime -> Seconds -> Bool
+isNightAllowanceApplicable nightShiftBounds tripStartTime now timeDiffFromUtc = do
+  let localRideEndDate = utctDay $ addUTCTime (secondsToNominalDiffTime timeDiffFromUtc) now
+      localTripStartTime = addUTCTime (secondsToNominalDiffTime timeDiffFromUtc) tripStartTime
+  case nightShiftBounds of
+    Nothing -> False
+    Just bounds -> do
+      let nightShiftStartTime = timeOfDayToDiffTime bounds.nightShiftStart
+          nightShiftEndTime = timeOfDayToDiffTime bounds.nightShiftEnd
+      if nightShiftStartTime <= 6 * 60 * 60 -- NS starting and ending on same date
```

Comment:
Why this 6 hours?

Reply:
This is to handle cases when night shift stars and ends on same day. I'm assuming if night shift starts between 0-6 Am then it will be same day otherwise it will be different days for start and end.